File: Serene/Serene.Script/Imports/ServerImports/Northwind.OrderForm.cs
using jQueryApi;
using Serenity;
using Serenity.ComponentModel;
using System;
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel;
using System.Runtime.CompilerServices;

namespace Serene.Northwind
{
    public partial class OrderForm : PrefixedContext
    {
        [InlineConstant] public const string FormKey = "Northwind.Order";

        public OrderForm(string idPrefix) : base(idPrefix) {}

        public CustomerEditor CustomerID { [InlineCode("{this}.w('CustomerID', Serene.Northwind.CustomerEditor)")] get; private set; }
        public DateEditor OrderDate { [InlineCode("{this}.w('OrderDate', Serenity.DateEditor)")] get; private set; }
        public DateEditor RequiredDate { [InlineCode("{this}.w('RequiredDate', Serenity.DateEditor)")] get; private set; }
        public LookupEditor EmployeeID { [InlineCode("{this}.w('EmployeeID', Serenity.LookupEditor)")] get; private set; }
        public OrderDetailsEditor DetailList { [InlineCode("{this}.w('DetailList', Serene.Northwind.OrderDetailsEditor)")] get; private set; }
        public DateEditor ShippedDate { [InlineCode("{this}.w('ShippedDate', Serenity.DateEditor)")] get; private set; }
        public LookupEditor ShipVia { [InlineCode("{this}.w('ShipVia', Serenity.LookupEditor)")] get; private set; }
        public DecimalEditor Freight { [InlineCode("{this}.w('Freight', Serenity.DecimalEditor)")] get; private set; }
        public StringEditor ShipName { [InlineCode("{this}.w('ShipName', Serenity.StringEditor)")] get; private set; }
        public StringEditor ShipAddress { [InlineCode("{this}.w('ShipAddress', Serenity.StringEditor)")] get; private set; }
        public StringEditor ShipCity { [InlineCode("{this}.w('ShipCity', Serenity.StringEditor)")] get; private set; }
        public StringEditor ShipRegion { [InlineCode("{this}.w('ShipRegion', Serenity.StringEditor)")] get; private set; }
        public StringEditor ShipPostalCode { [InlineCode("{this}.w('ShipPostalCode', Serenity.StringEditor)")] get; private set; }
        public StringEditor ShipCountry { [InlineCode("{this}.w('ShipCountry', Serenity.StringEditor)")] get; private set; }
    }
}

File: Serene/Serene.Core/Imports/ServerTypings/Northwind.OrderForm.ts
namespace Serene.Northwind {
    export interface OrderForm {
        CustomerID: CustomerEditor;
        OrderDate: Serenity.DateEditor;
        RequiredDate: Serenity.DateEditor;
        EmployeeID: Serenity.LookupEditor;
        DetailList: OrderDetailsEditor;
        ShippedDate: Serenity.DateEditor;
        ShipVia: Serenity.LookupEditor;
        Freight: Serenity.DecimalEditor;
        ShipName: Serenity.StringEditor;
        ShipAddress: Serenity.StringEditor;
        ShipCity: Serenity.StringEditor;
        ShipRegion: Serenity.StringEditor;
        ShipPostalCode: Serenity.StringEditor;
        ShipCountry: Serenity.StringEditor;
    }

    export class OrderForm extends Serenity.PrefixedContext {
        static formKey = 'Northwind.Order';
        private static init: boolean;

        constructor(prefix: string) {
            super(prefix);

            if (!OrderForm.init)  {
                OrderForm.init = true;

                var s = Serenity;
                var w0 = CustomerEditor;
                var w1 = s.DateEditor;
                var w2 = s.LookupEditor;
                var w3 = OrderDetailsEditor;
                var w4 = s.DecimalEditor;
                var w5 = s.StringEditor;

                Q.initFormType(OrderForm, [
                    'CustomerID', w0,
                    'OrderDate', w1,
                    'RequiredDate', w1,
                    'EmployeeID', w2,
                    'DetailList', w3,
                    'ShippedDate', w1,
                    'ShipVia', w2,
                    'Freight', w4,
                    'ShipName', w5,
                    'ShipAddress', w5,
                    'ShipCity', w5,
                    'ShipRegion', w5,
                    'ShipPostalCode', w5,
                    'ShipCountry', w5
                ]);
            }
        }
    }
}
